external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -170,10 +182,13 @@ private void initialize(Collection<TopicPartition> partitions) {
                 final long fetchOffset = doSeek(tp, committedOffset);
                 setAcked(tp, fetchOffset);
             }
-            initialized = true;
+
+            task.start();
```

Comment:
Since this is invoked from `onPartitionsAssigned`, the isComplete is going to return true as soon as onPartitionsRevoked is invoked after partition assignment right ?  

The `onPartitionsRevoked` could wait for message.timeout.secs so that in-flight messages are acked/failed and then commit the offsets before returning to minimize the duplicates. May be this is what you wanted the ConsumerRebalanceHandlerTask to handle via isCompleted but it does not seem to do that currently.